src/aligned_memory.rs

```diff
@@ -5,31 +5,31 @@ use std::mem;
 
 /// Provides u8 slices at a specified alignment
 #[derive(Clone, Debug, PartialEq)]
-pub struct AlignedMemory {
+pub struct AlignedMemory<const N: usize> {
     max_len: usize,
     align_offset: usize,
     mem: Vec<u8>,
 }
-impl AlignedMemory {
-    fn get_mem(max_len: usize, align: usize) -> (Vec<u8>, usize) {
-        let mut mem: Vec<u8> = Vec::with_capacity(max_len + align);
+impl<const N: usize> AlignedMemory<N> {
```

Comment:
Bad convention to use something more descriptive here like `ALIGN` rather than `N`?

Reply:
Heh I actually started with `ALIGN` then switched it to `N` since rust generics tend to be single capital letters. But I agree with you `ALIGN` is better i'll switch it back